src/Features/Core/Diagnostics/WorkspaceAnalyzerManager_HostDiagnostics.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics.Log;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal sealed partial class WorkspaceAnalyzerManager
```

Comment:
I really don't like adding all these static methods and event. this was never meant to used as helper. these should be an instance that passed around.

all these static looks like a hack.